pyproject.toml

```diff
@@ -5,11 +5,11 @@ requires = ["maturin>=1.5.0,<2.0.0"]
 [project]
 authors = [{name = "Eventual Inc", email = "daft@eventualcomputing.com"}]
 dependencies = [
-  "pyarrow >= 8.0.0",
-  "fsspec",
-  "tqdm",
-  "typing-extensions >= 4.0.0; python_version < '3.11'",
-  "packaging"
+  "pyarrow >= 8.0.0,<22.1.0",
+  "fsspec<2025.11.0",
+  "tqdm<4.68.0",
+  "typing-extensions >= 4.0.0,<4.16.0; python_version < '3.11'",
+  "packaging<25.1.0"
```

Comment:
Totally agree with this, but can we be a bit more flexible with some of these so we don't need to keep up as often, especially since we don't always keep up with dependabot. In particular:

* For PyArrow, can we limit to <23.0.0, accepting minor version upgrades? They usually don't have breaking changes
* No problem with fsspec, honestly I think we can make fsspec optional now right?
* TQDM doesn't update often anymore, so its fine
* I don't think `typing-extensions` needs a max, since its mainly intended for backporting features anyways
* Similar with packaging, I think it usually comes with `pip`, `setuptools`, `wheel` etc, so its very easy to run into version conflicts in virtualenvs. We only use 1 really basic feature of it anyways